sdk/keyvault/azure-security-keyvault-keys/src/key_client.cpp

```diff
@@ -130,6 +219,21 @@ Azure::Security::KeyVault::Keys::DeleteKeyOperation KeyClient::StartDeleteKey(
           {_detail::KeysPath, name}));
 }
 
+Azure::Security::KeyVault::Keys::RecoverDeletedKeyOperation KeyClient::StartRecoverDeletedKey(
```

Comment:
Does `StartXXXXX` match other languages?  RecoverDeletedKey is a more reasonable function name IMO.  If Start is consistent across languages we should stay with the pattern

Reply:
`Start` is what we use in .NET, but other languages use `begin`, IIRC. But `Begin` in .NET has the aire of older-style async operations, which is why we didn't use it. So use whatever is more idiomatic for C++ for async operations.

Reply:
...which I know C++ doesn't have natively until co_await is ratified. Or was it? I haven't been following lately.

Reply:
For LROs, I believe the pattern is to use `Start`, even in C+++
https://azure.github.io/azure-sdk/cpp_introduction.html#cpp-longrunning
![image](https://user-images.githubusercontent.com/6527137/112537216-a8ec2f00-8d6b-11eb-873f-3c9965a42120.png)

If we need to make an exception, we should include it in the C++ guideline